### PR TITLE
Update .travis.yml to run linker checker on all builds (failures allowed)

### PR DESCRIPTION
**Pull request by mit-mit**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,9 +29,13 @@ matrix:
       dart: dev
     - env: TASK="./tool/check-code.sh"
       dart: dev
+    - env: TASK="bundle exec jekyll build;./tool/shared/check-links.sh"
```
Comment (kwalrath) [inline .travis.yml]: Should this be `check-links.sh -e`?

Comment (mit-mit) [reply .travis.yml]: That's for checking external paths. What I'm doing here is ensuring that inside this matrix job the site is actually generated first. But I got the syntax wrong.

**Comment by mit-mit** [comment]
After (too) much experimentation, this appears to be working:
https://travis-ci.org/dart-lang/site-www/jobs/553694124

The failure in other jobs in the matrix is unrelated and is also happening in head.

I'd like to get the link checker to fail the check job on warnings, but need a change in shared for that:
https://github.com/dart-lang/site-shared/pull/101

**Comment by mit-mit** [comment]
@kevmoo can you help with reviewing this?

File: .travis.yml [hunk 2]
```diff
@@ -14,23 +13,28 @@ env:
...
+    - env: TASK="bundle exec jekyll build"
+      dart: stable
+    - env: TASK="./tool/check-code.sh"
+      dart: stable
+    - env: TASK="./tool/check-code.sh"
+      dart: dev
+    - env: TASK="./tool/analyze-and-test-examples.sh"
+      dart: stable
+    - env: TASK="./tool/analyze-and-test-examples.sh"
+      dart: dev
```
Comment (johnpryan) [inline .travis.yml]: should this be `stable`?

**Comment by kwalrath** [comment]
Thanks, @johnpryan!

**Pull request merged**